cli/src/commands/provider/gen.ts

```diff
@@ -15,33 +15,128 @@
+
 import { BaseCommand, baseFlags } from "../../baseCommand.js";
+import { withdrawFromNox } from "../../lib/chain/distributeToNox.js";
 import { commandObj } from "../../lib/commandObj.js";
 import { ensureComputerPeerConfigs } from "../../lib/configs/project/provider.js";
 import {
   CHAIN_FLAGS,
   PROVIDER_CONFIG_FULL_FILE_NAME,
   PROVIDER_SECRETS_CONFIG_FULL_FILE_NAME,
+  DOT_FLUENCE_DIR_NAME,
+  SECRETS_DIR_NAME,
+  BACKUPS_DIR_NAME,
+  ALL_FLAG_VALUE,
+  MAX_TOKEN_AMOUNT_KEYWORD,
+  CLI_NAME,
 } from "../../lib/const.js";
+import { pathExists, stringifyUnknown } from "../../lib/helpers/utils.js";
 import { initCli } from "../../lib/lifeCycle.js";
 import {
   getFluenceSecretsDir,
   ensureFluenceConfigsDir,
+  ensureProviderSecretsConfigPath,
+  getFluenceBackupsDir,
+  ensureDir,
 } from "../../lib/paths.js";
+import { confirm } from "../../lib/prompt.js";
+
+const RESET_NOX_SECRETS_FLAG_NAME = "reset-nox-secrets";
+const NO_WITHDRAW_FLAG_NAME = "no-withdraw";
 
 export default class Gen extends BaseCommand<typeof Gen> {
   static override description = `Generate Config.toml files according to ${PROVIDER_CONFIG_FULL_FILE_NAME} and secrets according to ${PROVIDER_SECRETS_CONFIG_FULL_FILE_NAME}`;
   static override examples = ["<%= config.bin %> <%= command.id %>"];
   static override flags = {
     ...baseFlags,
     ...CHAIN_FLAGS,
+    [RESET_NOX_SECRETS_FLAG_NAME]: Flags.boolean({
+      description: `Withdraw remaining tokens from your noxes, backup nox secrets from ${DOT_FLUENCE_DIR_NAME}/${PROVIDER_SECRETS_CONFIG_FULL_FILE_NAME} and ${DOT_FLUENCE_DIR_NAME}/${SECRETS_DIR_NAME} (if they exist) to ${DOT_FLUENCE_DIR_NAME}/${BACKUPS_DIR_NAME} and generate new ones`,
+      default: false,
+    }),
+    [NO_WITHDRAW_FLAG_NAME]: Flags.boolean({
+      description: `Is used only when --${RESET_NOX_SECRETS_FLAG_NAME} flag is present. Will not withdraw tokens from noxes (if you don't need it or it fails for some reason)`,
+      default: false,
+    }),
   };
   async run(): Promise<void> {
-    await initCli(this, await this.parse(Gen));
+    const { flags } = await initCli(this, await this.parse(Gen));
+    const providerSecretsConfigPath = await ensureProviderSecretsConfigPath();
+    const fluenceSecretsDir = getFluenceSecretsDir();
+
+    const backupDirPath = join(
+      getFluenceBackupsDir(),
+      `secrets-${new Date().toISOString().replaceAll(":", "")}`,
+    );
+
+    if (
+      flags[RESET_NOX_SECRETS_FLAG_NAME] &&
+      (await confirm({
+        message: `Are you sure you want to backup nox secrets ${color.yellow(providerSecretsConfigPath)} and ${color.yellow(fluenceSecretsDir)} (if they exist) to ${color.yellow(backupDirPath)} and generate new ones`,
+        default: flags[RESET_NOX_SECRETS_FLAG_NAME],
+      }))
+    ) {
+      if (
+        !flags[NO_WITHDRAW_FLAG_NAME] &&
+        (await confirm({
+          message:
+            "Do you want to withdraw remaining tokens from your noxes before continuing",
```

Comment:
shall there be "?" in the end?

Reply:
? is added automatically in the beggining of every prompt by the lib that is used in CLI so in CLI code ? is not needed to be written explicitly in any of the prompts